In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import plotly.graph_objects as go
import re
import folium
import geopandas as gpd
from folium.plugins import HeatMap
from folium.plugins import MarkerCluster 

In [2]:
# Importar archivo para trabajar
df_fran = pd.read_csv('/Users/benjaminzelaya/Desktop/PGF/PG/PF_DS_REVIEWS_AND_RECOMMENDATIONS/sprint_2/datasets_limpios/Franquicias_Inversion.csv')

In [3]:
df_fran

Id_Franquicia      Nombre_Franquicia  Min_Inversion  Max_Inversion  \
0                1                 Subway       190000.0       522000.0   
1                2             McDonald's       521000.0      2500000.0   
2                3                Dunkin'       121000.0      1810000.0   
3                4  Taco Bell Traditional       576000.0      3370000.0   
4                5            Burger King       232000.0      4520000.0   
..             ...                    ...            ...            ...   
934            935          Dazbog Coffee       187000.0       397000.0   
935            936               Vino 100        65000.0       431000.0   
936            937              Deli Zone       270000.0       437000.0   
937            938            O'Charley's      2100000.0      3140000.0   
938            939     Earthfruits Yogurt       258000.0       570000.0   

     Año_Fundado  Unidades  
0           1967     20576  
1           1955     13455  
2           1954      8118  
3           1962      7513  
4           1954      7043  
..           ...       ...  
934         1996         0  
935         2003         0  
936         2010         0  
937         1971         0  
938         2009         0  

[939 rows x 6 columns]

In [4]:
df_fran.shape

(939, 6)

In [5]:
df_fran.duplicated().sum()

0

In [5]:
df_fran.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 939 entries, 0 to 938
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Id_Franquicia      939 non-null    int64  
 1   Nombre_Franquicia  939 non-null    object 
 2   Min_Inversion      939 non-null    float64
 3   Max_Inversion      939 non-null    float64
 4   Año_Fundado        939 non-null    int64  
 5   Unidades           939 non-null    int64  
dtypes: float64(2), int64(3), object(1)
memory usage: 44.1+ KB


In [6]:
#proporciona una descripción rápida y útil de las estadísticas de la columna Max_Inversion	
df_fran['Max_Inversion'].describe()

count    9.390000e+02
mean     4.640620e+06
std      4.858272e+07
min      4.100000e+04
25%      4.570000e+05
50%      6.970000e+05
75%      1.320000e+06
max      9.776000e+08
Name: Max_Inversion, dtype: float64

In [7]:
#proporciona una descripción rápida y útil de las estadísticas de la columna Unidades	
df_fran['Unidades'].describe()

count      939.000000
mean       235.374867
std       1069.648008
min          0.000000
25%          7.000000
50%         23.000000
75%         86.000000
max      20576.000000
Name: Unidades, dtype: float64

In [8]:
#verificamos los valores unicos en la columna Nombre_Franquicia

df_fran['Nombre_Franquicia'].unique()

array(['Subway', "McDonald's", "Dunkin'", 'Taco Bell Traditional',
       'Burger King', 'Dominos', 'Bimbo Bakeries', 'Wendys',
       'Pizza Hut Traditional', 'Advanced Fresh Concepts',
       'Little Caesar', 'KFC Traditional', 'Sonic Drive-In', "Arby's",
       "Papa John's", 'Popeyes Louisiana Kitchen',
       "Noble Roman's Craft Pizza & Pub", "Jimmy John's",
       'Chick Fil A (Franchise Program)', "Jersey Mike's",
       'Panda Express', 'Jack In The Box',
       'Hissho Sushi/Oumi Sushi/Sushi with Gusto', 'Panera Bread',
       'DQ Grill & Chill', "Hardee's", 'Wingstop', 'IHOP', "Applebee's",
       'Kona Ice', 'Five Guys', "Denny's", 'Snowfox',
       'Pizza Hut Non-Traditional', "Chili's", 'Tropical Smoothie Cafe',
       'Buffalo Wild Wings', 'FIREHOUSE SUBS', "Papa Murphy's",
       "Auntie Anne's", "Carl's Jr.", 'Smoothie King', "Chester's",
       "Marco's Pizza", 'Baskin-Robbins', 'Cinnabon', "Church's Chicken",
       "Zaxby's", 'Cold Stone Creamery', 'Orion Food Syste

# ANALISIS EXPLORATORIO DE ESTADISTICAS DESCRIPTIVAS

In [9]:
# Calcular estadísticas descriptivas para columnas numéricas
descripcion_estadistica = df_fran.describe()

# Mostrar el resumen de estadísticas descriptivas
print(descripcion_estadistica)


       Id_Franquicia  Min_Inversion  Max_Inversion  Año_Fundado      Unidades
count     939.000000   9.390000e+02   9.390000e+02   939.000000    939.000000
mean      470.000000   2.208883e+06   4.640620e+06  2002.919063    235.374867
std       271.210251   2.461830e+07   4.858272e+07    18.373751   1069.648008
min         1.000000   3.000000e+03   4.100000e+04  1926.000000      0.000000
25%       235.500000   1.695000e+05   4.570000e+05  1995.000000      7.000000
50%       470.000000   2.890000e+05   6.970000e+05  2010.000000     23.000000
75%       704.500000   5.120000e+05   1.320000e+06  2016.000000     86.000000
max       939.000000   4.695000e+08   9.776000e+08  2023.000000  20576.000000


### Estas estadísticas descriptivas para las columnas numéricas en el conjunto de datos muestra lo siguiente:

Para la columna "Id_Franquicia", hay un total de 939 registros con valores que van desde 1 hasta 939.

La columna "Min_Inversion" tiene una inversión mínima de $3,000 y una inversión máxima de $469.5 millones, con una media de aproximadamente $2.21 millones y una desviación estándar de alrededor de $24.62 millones.

La columna "Max_Inversion" muestra una inversión máxima de $977.6 millones, con una inversión mínima de $41,000. La inversión promedio es de aproximadamente $4.64 millones, con una desviación estándar de alrededor de $48.58 millones.

En cuanto al año de fundación en la columna "Año_Fundado", el valor mínimo es 1926 y el máximo es 2023. La media es de aproximadamente 2002.92, con una desviación estándar de alrededor de 18.37 años.

Finalmente, la columna "Unidades" muestra el número de unidades que varía desde 0 hasta 20,576, con una media de aproximadamente 235.37 y una desviación estándar de alrededor de 1,069.65.

Estos estadísticas nos proporcionan una visión general de la distribución de los valores en estas columnas y pueden ser útiles para comprender mejor el conjunto de datos y tomar decisiones informadas en análisis posteriores.

In [10]:
# Obtener las 20 franquicias con más unidades
top_20_franquicias = df_fran.nlargest(20, 'Unidades')

# Crear el gráfico interactivo utilizando Plotly
fig = go.Figure()

# Agregar las barras
fig.add_trace(go.Bar(
    x=top_20_franquicias['Nombre_Franquicia'],
    y=top_20_franquicias['Unidades'],
    marker_color=top_20_franquicias['Año_Fundado'],
    text=top_20_franquicias['Año_Fundado'], 
    hoverinfo='x+y+text'
))

# Personalizar el diseño del gráfico
fig.update_layout(
    title='Año de Fundación vs. Cantidad de Sucursales (Top 20 Franquicias)',
    xaxis_title='Franquicia',
    yaxis_title='Cantidad de Sucursales',
    xaxis_tickangle=-45, 
    coloraxis_colorbar=dict(title='Año de Fundación')
)



Al analizar el gráfico que compara el año de fundación con la cantidad de sucursales de las 20 franquicias más grandes, se pueden extraer varias conclusiones significativas:

En primer lugar, se destaca que no existe una correlación lineal clara entre la antigüedad de una franquicia y su expansión. Aunque franquicias más antiguas muestran una presencia sólida en términos de sucursales, franquicias relativamente nuevas también ocupan posiciones destacadas. Este hecho sugiere que el éxito en la expansión de una franquicia no está determinado únicamente por su antigüedad, sino que factores como la estrategia de negocio, la demanda del mercado y la gestión eficiente pueden desempeñar roles fundamentales. 

Además, el gráfico refleja la diversidad en la distribución de franquicias, con algunas de larga data que siguen liderando y otras más recientes que han logrado un crecimiento rápido. En resumen, el gráfico ilustra que el mundo de las franquicias es complejo y que tanto la experiencia como la innovación pueden ser igualmente exitosas en el logro de una expansión significativa.